File: Dopamine.OobeModule/OobeModule.vb
Imports Microsoft.Practices.Prism.Modularity
Imports Microsoft.Practices.Prism.Regions
Imports Microsoft.Practices.Unity
Imports Dopamine.OobeModule.ViewModels
Imports Dopamine.OobeModule.Views

Public Class OobeModule
    Implements IModule

#Region "Variables"
    Private ReadOnly mRegionManager As IRegionManager
    Private mContainer As IUnityContainer
#End Region

#Region "Construction"
    Public Sub New(iRegionManager As IRegionManager, iContainer As IUnityContainer)
        Me.mRegionManager = iRegionManager
        Me.mContainer = iContainer
    End Sub
#End Region

#Region "Functions"
    Public Sub Initialize() Implements IModule.Initialize
        Me.mContainer.RegisterType(Of Object, OobeAppName)(GetType(OobeAppName).FullName)
        Me.mContainer.RegisterType(Of Object, OobeAppNameViewModel)(GetType(OobeAppNameViewModel).FullName)
        Me.mContainer.RegisterType(Of Object, OobeControls)(GetType(OobeControls).FullName)
        Me.mContainer.RegisterType(Of Object, OobeControlsViewModel)(GetType(OobeControlsViewModel).FullName)
        Me.mContainer.RegisterType(Of Object, OobeWelcome)(GetType(OobeWelcome).FullName)
        Me.mContainer.RegisterType(Of Object, OobeWelcomeViewModel)(GetType(OobeWelcomeViewModel).FullName)
        Me.mContainer.RegisterType(Of Object, OobeLanguage)(GetType(OobeLanguage).FullName)
        Me.mContainer.RegisterType(Of Object, OobeLanguageViewModel)(GetType(OobeLanguageViewModel).FullName)
        Me.mContainer.RegisterType(Of Object, OobeAppearance)(GetType(OobeAppearance).FullName)
        Me.mContainer.RegisterType(Of Object, OobeAppearanceViewModel)(GetType(OobeAppearanceViewModel).FullName)
        Me.mContainer.RegisterType(Of Object, OobeCollection)(GetType(OobeCollection).FullName)
        Me.mContainer.RegisterType(Of Object, OobeCollectionViewModel)(GetType(OobeCollectionViewModel).FullName)
        Me.mContainer.RegisterType(Of Object, OobeDonate)(GetType(OobeDonate).FullName)
        Me.mContainer.RegisterType(Of Object, OobeDonateViewModel)(GetType(OobeDonateViewModel).FullName)
        Me.mContainer.RegisterType(Of Object, OobeFinish)(GetType(OobeFinish).FullName)
        Me.mContainer.RegisterType(Of Object, OobeFinishViewModel)(GetType(OobeFinishViewModel).FullName)
    End Sub
#End Region
End Class

File: Dopamine.OobeModule/OobeModule.cs
using Dopamine.OobeModule.ViewModels;
using Dopamine.OobeModule.Views;
using Prism.Modularity;
using Prism.Regions;
using Microsoft.Practices.Unity;

namespace Dopamine.OobeModule
{
    public class OobeModule : IModule
    {
        #region Variables
        private readonly IRegionManager regionManager;
        private IUnityContainer container;
        #endregion

        #region Construction
        public OobeModule(IRegionManager regionManager, IUnityContainer container)
        {
            this.regionManager = regionManager;
            this.container = container;
        }
        #endregion

        #region IModule
        public void Initialize()
        {
            // Register Views and ViewModels with UnityContainer
            this.container.RegisterType<object, OobeAppName>(typeof(OobeAppName).FullName);
            this.container.RegisterType<object, OobeAppNameViewModel>(typeof(OobeAppNameViewModel).FullName);
            this.container.RegisterType<object, OobeControls>(typeof(OobeControls).FullName);
            this.container.RegisterType<object, OobeControlsViewModel>(typeof(OobeControlsViewModel).FullName);
            this.container.RegisterType<object, OobeWelcome>(typeof(OobeWelcome).FullName);
            this.container.RegisterType<object, OobeWelcomeViewModel>(typeof(OobeWelcomeViewModel).FullName);
            this.container.RegisterType<object, OobeLanguage>(typeof(OobeLanguage).FullName);
            this.container.RegisterType<object, OobeLanguageViewModel>(typeof(OobeLanguageViewModel).FullName);
            this.container.RegisterType<object, OobeAppearance>(typeof(OobeAppearance).FullName);
            this.container.RegisterType<object, OobeAppearanceViewModel>(typeof(OobeAppearanceViewModel).FullName);
            this.container.RegisterType<object, OobeCollection>(typeof(OobeCollection).FullName);
            this.container.RegisterType<object, OobeCollectionViewModel>(typeof(OobeCollectionViewModel).FullName);
            this.container.RegisterType<object, OobeDonate>(typeof(OobeDonate).FullName);
            this.container.RegisterType<object, OobeDonateViewModel>(typeof(OobeDonateViewModel).FullName);
            this.container.RegisterType<object, OobeFinish>(typeof(OobeFinish).FullName);
            this.container.RegisterType<object, OobeFinishViewModel>(typeof(OobeFinishViewModel).FullName);
        }
        #endregion
    }
}
